tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h

```diff
@@ -119,8 +124,24 @@ template <>
 inline Value MapLhloOpToStdScalarOp<xla_lhlo::AbsOp>(
     Location loc, ArrayRef<Type> result_types, ArrayRef<Value> args,
     OpBuilder* b) {
-  return MapLhloOpToStdScalarOpImpl<FloatType, ::mlir::AbsFOp>{}(
-      loc, result_types, args, b);
+  Type element_type = args.front().getType();
+  if (element_type.isa<FloatType>()) {
+    return MapLhloOpToStdScalarOpImpl<FloatType, ::mlir::AbsFOp>{}(
+        loc, result_types, args, b);
+  }
+  if (element_type.isa<IntegerType>()) {
+    const auto& lhs = args[0];
+    auto integer_type = element_type.dyn_cast<IntegerType>();
+
+    auto zero_intval = b->create<::mlir::ConstantIntOp>(loc, 0, integer_type.getWidth());
+    auto lhs_gt_zero = b->create<ScalarIOp<CompareOp>>(
+        loc, CmpIPredicate::sge, lhs, zero_intval);
+    auto neg_val =  b->create<ScalarIOp<xla_lhlo::NegOp>>(
+        loc, result_types,
+        args, mlir::None);
```

Comment:
Seems like we can just use `sub(zero_intval, lhs)` instead of the NegOp?

Reply:
Ok mehdi ,I'll update it